File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_nothink/20260910_094114/python/008_prime_numbers/solution.py
import sys

def is_prime(n):
    """Checks if a number is prime."""
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    # Check odd divisors up to the square root of n
    for i in range(3, int(n**0.5) + 1, 2):
        if n % i == 0:
            return False
    return True

def main():
    # Ensure an argument was provided
    if len(sys.argv) < 2:
        return

    try:
        # Parse the first command line argument as an integer
        n = int(sys.argv[1])
    except ValueError:
        # If the argument is not a valid integer, exit silently or handle error
        return

    # Iterate from 2 up to and including N
    for num in range(2, n + 1):
        if is_prime(num):
            print(num)

if __name__ == "__main__":
    main()

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_nothink/20260910_094114/go/008_prime_numbers/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

// isPrime checks if a number is prime.
func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 || n == 3 {
		return true
	}
	if n%2 == 0 || n%3 == 0 {
		return false
	}
	// Use 6k +/- 1 rule to optimize primality test
	for i := 5; i*i <= n; i += 6 {
		if n%i == 0 || n%(i+2) == 0 {
			return false
		}
	}
	return true
}

func main() {
	// Check if the command line argument N is provided
	if len(os.Args) < 2 {
		return
	}

	// Parse the argument to an integer
	n, err := strconv.Atoi(os.Args[1])
	if err != nil {
		// If input is not a valid integer, we exit silently or could print an error.
		// Based on the prompt requirements, we focus on printing primes.
		return
	}

	// Iterate from 2 up to N and print if the number is prime
	for i := 2; i <= n; i++ {
		if isPrime(i) {
			fmt.Println(i)
		}
	}
}